frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -133,6 +138,33 @@ export const GRAPH_DATA_SETTINGS = {
     dashboard: false,
     useRawSeries: true,
   },
+  series_order: {
+    section: t`Data`,
+    widget: ChartSettingOrderedSimple,
+    getDefault: series => {
+      const keys = series.map(s => keyForSingleSeries(s));
+      return keys.map((key, index) => ({
+        name: key,
+        originalIndex: index,
+        enabled: true,
+      }));
+    },
+    getProps: series => {
+      const keys = series.map(s => keyForSingleSeries(s));
+      return {
+        items: keys.map((key, index) => ({
+          name: key,
+          originalIndex: index,
+        })),
+      };
+    },
+    getHidden: ({ _raw, ...series }, settings) => {
+      return (
+        settings["graph.dimensions"]?.length < 2 ||
+        Object.keys(series).length > 20
```

Comment:
I guess, correct me if I'm wrong since this is so much new to me haha, `_raw` is set from https://github.com/metabase/metabase/blob/1c321aa4969224e94e74944892468ceb2421a403/frontend/src/metabase/visualizations/index.js#L77-L78

That means the underlying object is still an array, so there shouldn't be a need to destructure `_raw`, if you get the whole parameter as `series` you should still be able to do `series.length` directly.

Can you try that out? I think this will look easier to understand.

```suggestion
    getHidden: (series, settings) => {
      return (
        settings["graph.dimensions"]?.length < 2 ||
        series.length > 20
```

Reply:
Ahh. I thought that would mess with the length property. I can simplify this, yeah